app/validators/routing/when_rule_validator.py

```diff
@@ -0,0 +1,188 @@
+from app.validators.validator import Validator
+
+
+class WhenRuleValidator(Validator):
+    LIST_REFERENCE_INVALID = "Invalid list reference"
+    CHECKBOX_MUST_USE_CORRECT_CONDITION = (
+        "The condition cannot be used with `Checkbox` answer type"
+    )
+    CHECKBOX_CONDITION_ONLY = (
+        "This condition can only be used with `Checkbox` answer types"
+    )
+    NON_CHECKBOX_COMPARISON_ID = (
+        "The comparison id is not of answer type `Checkbox`. The condition can only reference "
+        "`Checkbox` answers when using `comparison id`"
+    )
+    NON_MATCHING_WHEN_ANSWER_AND_COMPARISON_TYPES = (
+        "The answers used as comparison id and answer id in the `when` clause have "
+        "different types"
+    )
+    INVALID_WHEN_RULE_ANSWER_VALUE = "Answer value in when rule has an invalid value"
+    NON_EXISTENT_WHEN_KEY = (
+        'The answer id in the key of the "when" clause does not exist'
+    )
```

Comment:
I think we should make references in the error messages consistent. Some referenced named keys/values using double quotes, some use backticks, and some don't use quotes at all.

Reply:
I agree, but I think this is part of smaller changes we make as part of continuing to improve validator.